docs/components.md

```diff
@@ -21,28 +21,44 @@ KubeVirt consists of a set of services:
-                     |          |         v                            |
-                     |          |      POST /pods ----> validate       |
-                     |          |         |             create         |
-                     |          |         |             schedPod -> observe
-                     |          |         |                            v
-                     | <------------------------------------------> GET /vms
-                     |          |         |                         defineVM
-                     |          |         |                         watchVM
-                     |          |         |                            |
-    DELETE /vms -> validate     |         |                            |
-                   delete ----> * --------------------------------> observe
-                     |                    |                         shutdownVM
-                     |                    |                            |
-                     :                    :                            :
+```
+Client         Virt API    VM TPR  Virt Controller  k8s         VM Handler
+-------------- ----------- ------- ---------------- ----------- ----------
+
+               listen <----------- WATCH /vms
+               listen <---------------------------------------- WATCH /vms
+                                      |                            |
+POST /vms ---> validate               |                            |
+               create ---> VM ---> observe                         |
+                 |          |         v                            |
+                 |          |      POST /pods ----> validate       |
+                 |          |         |             create         |
+                 |          |         |             schedPod -> observe
+                 |          |         |                            v
+                 | <------------------------------------------> GET /vms
+                 |          |         |                         defineVM
+                 |          |         |                            |
+                 |          |         |                            |
+DELETE /vms -> validate     |         |                            |
+               delete ----> * --------------------------------> observe
+                 |                    |                         shutdownVM
+                 |                    |                            |
+                 :                    :                            :
+```
+
+1. A client posts a new VM definition to the `virt-api-server`
+2. The `virt-api-server` validates the input and creates a `VM` 3rd party
+   resource (TPR) object through the Kubernetes API Server
+3. The `virt-controller` observes the creation of the new `VM` object
+   and creates a corrsponding pod.
+4. Kubernetes is scheduling the pod on a host
+5. The `virt-handler` (_DaemonSet_) observes that a pod for a `VM` got
```

Comment:
This point is not correct. The rough description is:

 * virt-controller observes that a pod for a VM got started and updates the VM TPR with a `nodeName` label. This update is what hands over the responsibility from virt-controller to virt-handler.

 * virt-handler can therefore only watches VMs with a nodeName label equal to the node where virt-handler is running on. So it gets the VM spec through observation and not through explicit fetching and it only has to watch pods.